cordapp-cpk2/src/main/java/net/corda/plugins/cpk2/CordappPlugin.java

```diff
@@ -510,11 +512,17 @@ private void configureCordappTasks(@NotNull Project project) {
 
     private void configureCordappAttributes(@NotNull String symbolicName, @NotNull Attributes attributes) {
         attributes.put(BUNDLE_SYMBOLICNAME, symbolicName);
-        attributes.put(CPK_CORDAPP_NAME, symbolicName);
         attributes.put(CPK_FORMAT_TAG, CPK_FORMAT);
 
         final CordappData contract = cordapp.getContract();
+        Pattern cordappCpkNamePattern = Pattern.compile("[a-zA-Z0-9.\\-]+");
         if (!contract.isEmpty()) {
+            final String cordappCpkName = getCordaCpkCordappName(contract);
+            if (cordappCpkName != null) {
+                attributes.put(CPK_CORDAPP_NAME, cordappCpkName);
+            } else {
+                attributes.put(CPK_CORDAPP_NAME, symbolicName);
+            }
```

Comment:
Think we could also bake into the function this piece? Then it becomes something like `populateCordappCpkName`